cf-custom-resources/lib/wkld-cert-validator.js

```diff
@@ -365,8 +365,9 @@ async function requestCertificate({ aliases, idempotencyToken }) {
  * @param {Set<String>} aliases for the service.
  */
 async function waitForValidationOptionsToBeReady(certificateARN, aliases) {
-  let expectedCount = aliases.size + 1; // Expect one validation option for each alias and the cert domain.
-
+  // If the certificate domain is one of the aliases, expect one validation option for each alias.
+  // Otherwise, include an extra validation option for the certificate domain itself.
+  let expectedCount = aliases.has(certificateDomain) ? aliases.size : aliases.size + 1;
```

Comment:
Looks like we have the same problem with NLB as well: if they specify `${serviceName}-nlb.${envName}.${appName}.${domainName}` as the NLB alias 🤔

Reply:
yes that's true, we are using the same custom resource for nlb alias as well.